.circleci/config.yml

```diff
@@ -4430,6 +4430,31 @@ jobs:
           command: |
             ./release/scripts/vuln_check.sh || ./release/scripts/vuln_check.sh
 
+  scan-high-critical-vulnerabilities:
+    executor: custom
+    resource_class: small
+    steps:
+    - checkout
+    - setup_remote_docker
+    - run:
+        name: Install trivy
```

Comment:
This should be baked in CI image

Reply:
There's trivy image https://hub.docker.com/r/aquasec/trivy maybe just use that? No need to `curl | sh` which they call _insecure_ (https://aquasecurity.github.io/trivy/v0.27.1/docs/advanced/container/embed-in-dockerfile/).
The dockerhub page also tells how to cache vuln definitions between runs.

Reply:
Perhaps I'm not following your idea. You can use machine executor in CI that allows `docker run` commands. There run something like:
```sh
docker run -v /var/run/docker.sock:/var/run/docker.sock -v trivy-cache:/root/.cache/ aquasec/trivy image quay.io/rhacs-eng/main:$(make tag)
```

If you prefer docker-in-docker (I would prefer it), here are instructions https://circleci.com/docs/2.0/building-docker-images/ . The command like I suggested should work too.